### ISSUE TITLE
Client calculating incorrect total and free disk space when executable and --dir on different partitions or disks

### ISSUE THREAD

**Issue by bjacobowski** [BOINC/boinc#2325]
OS: arch
boinc version: 7.8.4

I've got limited space on my root partition, so I moved the data directory to a partition with more space, fired up boinc (via systemd or manually)
```bash
$ /usr/bin/boinc_client --dir /otherdir/boinc
```
and...no change in reported space. 

[HOST_INFO](https://github.com/BOINC/boinc/blob/master/lib/hostinfo.h) is responsible for getting/tracking available space, which it gets from [get_filesystem_info](https://github.com/BOINC/boinc/blob/e3500afee2390f9d3d75364404856372773d527d/lib/filesys.h#L95). 
```cpp
extern int get_filesystem_info(double& total, double& free, char* path=const_cast<char *>("."));
```
It's only ever called without providing a path (because HOST_INFO doesn't track the data directory and thus can't pass the arg). Same call signature in client/hostinfo_win.cpp and client/hostinfo_unix.cpp.
```cpp
int HOST_INFO::get_host_info(bool init) {
    int retval = get_filesystem_info(d_total, d_free);
```

**Comment by davidpanderson** [comment]
I don't understand this.  When the client parses the --dir argument, it immediately does a chdir() to that directory; this is done before it does get_host_info() or any other initialization.

**Comment by CharlieFenton** [comment]
what OS is "arch"?

**Comment by davidpanderson** [comment]
I was unable to repro this.  I ran the client with --dir /tmp and it used the disk space on that partition.

**Comment by Ageless93** [comment]
@CharlieFenton, https://www.archlinux.org/

**Comment by bjacobowski** [comment]
I apologize, the problem is on my end. I was having permissions issues getting boinc client to start in my home folder. In an effort to fix that, I created the /home/boinc folder, started there; what I failed to notice is that /home/boinc had been created on the / partition instead of the larger /home/<user> partition. I should have tried this in a clean VM before posting the issue, sorry for the time you spent looking into this.
